commit d306047d90177900018547610209f2c290794d10
Author: jb <noobinet@gmail.com>
Date:   Wed Mar 3 01:03:03 2010 +0100

    Fix the empty directory view bug.
    
    Dunno at all if it's ok to do that, but at least it works

diff --git a/src/mpddirectory.cpp b/src/mpddirectory.cpp
index f4ded77..79fa7b0 100644
--- a/src/mpddirectory.cpp
+++ b/src/mpddirectory.cpp
@@ -59,7 +59,7 @@ MPDDirectory::MPDDirectory(const MPDDirectory &other) : d(other.d) {}
 MPDDirectory::~MPDDirectory() {}
 
 bool MPDDirectory::isNull() const {
-	return d == NULL;
+	return !d;
 }
 
 MPDDirectoryPointer *MPDDirectory::pointer() const {
diff --git a/src/mpddirectory.h b/src/mpddirectory.h
index a7e4cfd..6042187 100644
--- a/src/mpddirectory.h
+++ b/src/mpddirectory.h
@@ -55,7 +55,7 @@ public:
 	MPDDirectory &operator=(const MPDDirectory &);
 
 private:
-	QSharedDataPointer<MPDDirectoryPrivate> d;
+	QExplicitlySharedDataPointer<MPDDirectoryPrivate> d;
 };
 
 class MPDDirectoryPointer {
